spec/Functions.tex

```diff
@@ -764,31 +769,45 @@ \subsection{The Ref Return Intent}
 \end{chapel}
 This code outputs the number \chpl{3}.
 
+\begin{chapeloutput}
+3
+\end{chapeloutput}
+\end{chapelexample}
+
+\begin{chapelexample}{ref-return-intent-pair.chpl}
+
 \index{setter}
 \index{functions!setter argument}
-The implicit \chpl{setter} argument can be used to ensure, for
+The getter-setter pair can be used to ensure, for
 example, that the second element in the pseudo-array is only assigned
-a value if the first argument is positive.  To do this, add the
-following:
-\begin{chapelnoprint}
-proc A2(i: int) ref { // } to fool latex
+a value if the first argument is positive.  The following is an
+example:
+\begin{chapel}
+var x, y = 0;
+
+proc doA(param setter, i: int) ref {
   if i < 0 || i > 1 then
     halt("array access out of bounds");
-\end{chapelnoprint}
-\begin{chapel}
-if setter && i == 1 && x <= 0 then
-  halt("cannot assign value to A(1) if A(0) <= 0");
-\end{chapel}
-\begin{chapelpost}
-// { to fool latex
+
+  if setter && i == 1 && x <= 0 then
+    halt("cannot assign value to A(1) if A(0) <= 0");
+
   if i == 0 then
     return x;
   else
     return y;
 }
-A2(0) = 0;
-A2(1) = 1;
-\end{chapelpost}
+proc A(i: int) ref {
+  return doA(true, i);
+}
+proc A(i: int) {
+  return doA(false, i);
+}
+
+A(0) = 0;
+A(1) = 1;
+
+\end{chapel}
 \begin{chapelprediff}
 \#!/usr/bin/env python
```

Comment:
Now that there is only one line to sort, can you be so cool as to remove this prediff?